configure.ac

```diff
@@ -443,11 +443,33 @@ if test "x$with_dl" != "xno"; then
   AC_CHECK_LIB(dl, dlopen)
 fi
 
+AC_DEFUN([LIBATOMIC_CHECKER],[
+  AC_MSG_CHECKING(whether linking $1 -latomic works)
+  AC_LINK_IFELSE([
+    AC_LANG_SOURCE([[
+      #include <atomic>
+      struct LockRequired { char data[256]; };
+      int
+      main() {
+          return std::atomic<LockRequired>{}.is_lock_free();
+      }
+  ]])],[
+    AC_MSG_RESULT(yes)
+    libatomic_checker_result="yes"],[
+    AC_MSG_RESULT(no)
+    libatomic_checker_result="no"
+])])
+
 ## check for atomics library before anything that might need it
-# AC_SEARCH_LIBS pollutes LIBS
 SQUID_STATE_SAVE(LIBATOMIC)
-AC_SEARCH_LIBS([__atomic_load_8],[atomic],[
-  test "$ac_res" = "none required" || ATOMICLIB=$ac_res],[])
+LIBATOMIC_CHECKER(without)
+if test "x$libatomic_checker_result" = "xno"; then
+  LIBS="$LIBS -latomic"
+  LIBATOMIC_CHECKER(with)
+  if test "x$libatomic_checker_result" = "xyes"; then
```

Comment:
Please consider AS_IF macro instead of the shell conditions. I do not insist on this, but it would make the autoconf syntax a bit cleaner.

Reply:
Done.